src/internal/error.ts

```diff
@@ -35,4 +36,15 @@ export class AWSError extends Error {
         const doc = parseHTML(xmlDocument)
         return new AWSError(doc.find('Message').text(), doc.find('Code').text())
     }
+
+    static parse(response: Response): AWSError {
+        if (response.headers['Content-Type'] === 'application/json') {
+            const error = response.json() as any;
+            const message = error.Message || error.message || error.__type || 'An error occurred on the server side';
+            const code = response.headers['X-Amzn-Errortype'] || error.__type
+            return new AWSError(message, code)
```

Comment:
Pulled some of the error handling logic that @jakub-qg into here to make it more reusable.